src/code/m_lib.c

```diff
@@ -0,0 +1,763 @@
+#include "m_lib.h"
+#include "game.h"
+#include "libc/math.h"
+#include "m_rcp.h"
+#include "macros.h"
+#include "sys_math.h"
+#include "sys_math_atan.h"
+
+void (*D_80107B00_jp[])(u8*, ValueSet*) = {
+    ValueSet__s_char, ValueSet__u_char,      ValueSet__s_short, ValueSet__u_short,     ValueSet__s_int, ValueSet__u_int,
+    ValueSet__float,  ValueSet__float_x1000, ValueSet__xyz_t,   ValueSet__xyz_t_x1000, ValueSet__s_xyz,
+};
+
+void mem_copy(u8* dest, u8* src, size_t len) {
+    while (len != 0) {
+        *dest = *src;
+        src++;
+        dest++;
+        len--;
+    }
+}
+
+void mem_clear(u8* ptr, size_t len, u8 value) {
```

Comment:
memory functions usually take a `void*` pointer and use a temp to cast to `u8*`, which makes the function simpler to use.
There's no room on left on the stack for a temp cast?

Reply:
The temps break the function